app/share.tsx

```diff
@@ -90,115 +71,68 @@ export const App = ({ root }: any) => (
 	</Stack.Navigator>
 );
 
-class Root extends React.Component<{}, IState> {
-	private mounted = false;
-
-	constructor(props: any) {
-		super(props);
-		const { width, height, scale, fontScale } = Dimensions.get('screen');
-		const theme = initialTheme();
-		this.state = {
-			theme: getTheme(theme),
-			themePreferences: theme,
-			root: '',
-			width,
-			height,
-			scale,
-			fontScale
-		};
-		this.init();
-	}
+const { width, height, scale, fontScale } = Dimensions.get('screen');
 
-	componentDidMount() {
-		this.mounted = true;
-	}
+const Root = () => {
+	const [root, setRoot] = useState('');
+	const theme = getTheme(initialTheme());
```

Comment:
While this is ok on React.Component's `constructor`, it's not on a function component.
It's going to re-run these functions on every render.
You should do it outside the component, as you did for Dimensions.

If you compare this logic with `App/Index`, you'll notice you did this:
```jsx
const [theme, setTheme] = useState(getTheme(initialTheme()));
```

This is totally fine, since you're running the functions only once for the default value of `useState()`.
Here, you can do it outside the component, since we don't really care if the theme changes inside the share extension

Reply:
Sure, You are right!